### PR TITLE
Minor improvements to sample.

### PR DESCRIPTION
**Pull request by Pushpalanka**

### CODE REVIEW

File: modules/samples/oauth2/custom-grant/src/main/java/org/wso2/sample/identity/oauth2/grant/mobile/MobileGrant.java [hunk 1]
```diff
@@ -154,4 +156,8 @@ private boolean isValidMobileNumber(String mobileNumber){
         return false;
     }
 
+    public boolean isOfTypeApplicationUser() throws IdentityOAuth2Exception {
```
Comment (pulasthi7) [inline modules/samples/oauth2/custom-grant/src/main/java/org/wso2/sample/identity/oauth2/grant/mobile/MobileGrant.java]: Do we need to make this method public?

Comment (Pushpalanka) [inline modules/samples/oauth2/custom-grant/src/main/java/org/wso2/sample/identity/oauth2/grant/mobile/MobileGrant.java]: This is an overridden method which has 'public' access modifier in super class.

File: modules/samples/oauth2/custom-grant/src/main/java/org/wso2/sample/identity/oauth2/grant/mobile/MobileGrant.java [hunk 2]
```diff
@@ -154,4 +156,9 @@ private boolean isValidMobileNumber(String mobileNumber){
         return false;
     }
 
+    @Override
+    public boolean isOfTypeApplicationUser() throws IdentityOAuth2Exception {
+        return true;
```
Comment (pulasthi7) [inline modules/samples/oauth2/custom-grant/src/main/java/org/wso2/sample/identity/oauth2/grant/mobile/MobileGrant.java]: Do we need that as the super class returns true by default?

Comment (Pushpalanka) [inline modules/samples/oauth2/custom-grant/src/main/java/org/wso2/sample/identity/oauth2/grant/mobile/MobileGrant.java]: Is there any harm keeping this?  Even if superclass later change this value, it won't effect as we are explicitly setting it here and in one look the details are visible within the class if we keep it. Anyway I can remove if there is a thorough objection to keep it.. :)

**Pull request merged**